ote/src/cljc/ote/transit_changes.cljc

```diff
@@ -309,8 +313,28 @@
    [25 12] :xmas-day
    [26 12] :boxing-day})
 
-(defn is-holiday?
-  "Check if given date is a holiday to be skipped. Returns holiday id if it is or nil otherwise."
-  [date]
-  (let [{::time/keys [month date]} (time/date-fields date)]
-    (static-holidays [date month])))
+#?(:clj
+   (defn exception-holidays
+    "returns a map like {[day month year] :holiday} based on detection-holidays table"
+    [db]
+    (let [dates (specql/fetch db
+                              :gtfs/detection-holidays
+                              #{:gtfs/date :gtfs/reason}
+                              {:gtfs/date op/not-null?})]
+      (reduce
+        (fn [map holiday]
+          (let [{::time/keys [month date year]} (time/date-fields (:gtfs/date holiday))]
+            (assoc map
+              [date month year] (keyword (str/lower-case (:gtfs/reason holiday))))))
+        {}
+        dates))))
+
+#?(:clj
+   (defn is-holiday?
+     "Check if given date is a holiday to be skipped. Returns holiday id if it is or nil otherwise."
+     [db date]
+     (let [{::time/keys [month date year]} (time/date-fields date)
+           exceptions (exception-holidays db)]
+       (if (exceptions [date month year])
```

Comment:
Two calls to `exceptions`, could be made more efficient binding the value using let or rather, `if-let`:

```
(if-let [res (exceptions [date month year])]
    res
   (static-holidays [date month]))